core/src/main/scala/org/http4s/Header.scala

```diff
@@ -182,7 +182,7 @@ object Header {
     type Aux[A, G[_]] = Select[A] { type F[B] = G[B] }
 
     def fromRaw[A](h: Header.Raw)(implicit ev: Header[A, _]): Option[Ior[ParseFailure, A]] =
-      (h.name == Header[A].name).guard[Option].as(Header[A].parse(h.value).toIor)
+      (h.name == Header[A].name).guard[Option].map(_ => Header[A].parse(h.value).toIor)
```

Comment:
this was the quickest change, but i wouldn't be against rewriting this to be an if statement so the laziness is more apparent. or we can add a comment